device/transport/http/devdoc/http_requirements.md

```diff
@@ -116,26 +117,101 @@ the getReceiver method queries the client for an `HttpReceiver` object that can
 
 **SRS_NODE_DEVICE_HTTP_16_011: [** The HTTP transport should use the x509 settings passed in the `options` object to connect to the service if present. **]**
 
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_001: [** The setOptions method shall accept an argument formatted as such:
+ {
+     interval: (Number),
+     at: (Date)
+     cron: (string)
+     drain: (Boolean)
+     manualPolling: (Boolean)
+}
+**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_008: [**Only one of the `interval`, at, `manualPolling` and `cron` fields should be populated: if more than one is populated, an ArgumentError shall be thrown.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_021: [**If `opts.interval` is set, messages should be received repeatedly at that interval**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_002: [**If `opts.interval` is not a number, an ArgumentError should be thrown.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_005: [**If `opts.interval` is a negative number, an ArgumentError should be thrown.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_003: [**if `opts.at` is set, messages shall be received at the Date and time specified.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_022: [**If `opts.at` is not a Date object, an ArgumentError should be thrown**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_020: [**If `opts.cron` is set messages shall be received according to the schedule described by the expression.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_004: [**if `opts.cron` is set it shall be a string that can be interpreted as a cron expression**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_007: [**If `opts.cron` is not a parseable cron string, an ArgumentError should be thrown.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_017: [**If `opts.drain` is true all messages in the queue should be pulled at once.**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_018: [**If `opts.drain` is false, only one message shall be received at a time**]**
+
+**SRS_NODE_DEVICE_HTTP_RECEIVER_16_019: [**If the receiver is already running with a previous configuration, the existing receiver should be restarted with the new configuration**]**
```

Comment:
>f the receiver is already running with a previous configuration, the existing receiver should be restarted with the new configuration [](start = 44, length = 133)

how?

Reply:
ah, never mind.  When  I first read this, I thought it was a requirement that you were placing on the consumer, but now I realize it's the responsibility of the receiver to restart itself.

---
In reply to: [143036839](https://github.com/Azure/azure-iot-sdk-node/pull/102#discussion_r143036839) [](ancestors = 143036839)